kernel/src/history_manager/mod.rs

```diff
@@ -0,0 +1,4 @@
+#[cfg(feature = "internal-api")]
+pub mod search;
+#[cfg(not(feature = "internal-api"))]
+pub(crate) mod search;
```

Comment:
maybe just leave as only `pub(crate)` unless we have a reason to expose as `internal-api`?

Reply:
(applies elsewhere too)